src/Libraries/DSCPython/CPythonEvaluator.cs

```diff
@@ -38,55 +40,78 @@ public static class CPythonEvaluator
+                            scope.Set((string)bindingNames[i], InputMarshaler.Marshal(bindingValues[i]).ToPython());
+                        }
 
-                            try
-                            {
-                                OnEvaluationBegin(scope, code, bindingValues);
-                                scope.Exec(code);
-                                OnEvaluationEnd(false, scope, code, bindingValues);
+                        try
+                        {
+                            OnEvaluationBegin(scope, code, bindingValues);
+                            scope.Exec(code);
 
-                                var result = scope.Contains("OUT") ? scope.Get("OUT") : null;
+                            var result = scope.Contains("OUT") ? scope.Get("OUT") : null;
 
-                                return OutputMarshaler.Marshal(result);
+                            return OutputMarshaler.Marshal(result);
+                        }
+                        catch (Exception e)
+                        {
+                            var traceBack = GetTraceBack(e);
+                            if (!string.IsNullOrEmpty(traceBack))
+                            {
+                                // Throw a new error including trace back info added to the message
+                                throw new DynamoException($"{e.Message} {traceBack}", e);
                             }
-                            catch (Exception e)
+                            else
                             {
-                                OnEvaluationEnd(false, scope, code, bindingValues);
-                                throw;
+                                throw e;
                             }
                         }
+                        finally
+                        {
+                            OnEvaluationEnd(false, scope, code, bindingValues);
+                        }
                     }
                 }
-                catch (PythonException pe)
-                {
-                    throw;
-                }
-                finally
-                {
-                    PythonEngine.ReleaseLock(gs);
-                }
+            }
+            finally
+            {
+                PythonEngine.ReleaseLock(gs);
+            }
+        }
+
+        /// <summary>
+        /// Gets the trace back message from the exception, if it is a PythonException.
+        /// </summary>
+        /// <param name="e">Exception to inspect</param>
+        /// <returns>Trace back message</returns>
+        private static string GetTraceBack(Exception e)
+        {
+            var pythonExc = e as PythonException;
+            if (!(e is PythonException))
+            {
+                return null;
+            }
+
+            // Return the value of the trace back field (private)
+            return typeof(PythonException).GetField("_tb", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(pythonExc) as string;
```

Comment:
should this be wrapped in its own try catch?

Reply:
I thought about it, but decided not to do that. Whenever that happens it would be because we have updated Python.NET and that fields no longer exists. Should we try to hide that under the rug or let it show up so that we know we need to update this code?

Reply:
we should definitely not crash....
I can think of a few options
* in the python node send a message asking the user to report
* send an event to analytics with an error which we can look for
* open something like a VM crash dialog - dynamo doesn't actually crash, but theres a message for the user to post the given log to github.

Reply:
I don't think an exception at this point would make the application crash. I said show up not blow up :)
Let me give it try and I'll attach a screenshot of how it would eventually look like.

Reply:
Well, it does not crash, but the node just gives a generic `Object reference not set to an instance of an object.` warning. I guess I'll throw a `DynamoException` with a special message to contact us :)